### PR TITLE
feature(frontend): prepare FE for granular data store service update

### PR DESCRIPTION
**Pull request by xoscar**
This PR updates the openAPI specs to now use a custom config endpoint just for data stores.
It also removes a lot of the FE logic to handle the full configuration object for the settings page.

## Changes

- Updates openAPI
- Removes unnecessary reducer and code

## Fixes

- https://github.com/kubeshop/tracetest/issues/1656

## Checklist

- [x] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

### CODE REVIEW

File: api/config.yaml [hunk 1]
```diff
@@ -53,6 +53,15 @@ components:
...
           $ref: "#/components/schemas/SignalFX"
+    UpdateDataStoreConfigRequest:
+      type: object
+      properties:
+        dataStores:
+          type: array
+          items:
+            $ref: "#/components/schemas/DataStore"
+        defaultDataStore:
+          $ref: "#/components/schemas/SupportedDataStores"
```
Comment (mathnogueira) [inline api/config.yaml:64]: I think this should be a string instead. The user might have 2 different jaeger instances and want to switch between them. This should reference the data store id

File: api/openapi.yaml [hunk 2]
```diff
@@ -1073,24 +1073,23 @@ paths:
             application/json:
               schema:
                 $ref: "./config.yaml#/components/schemas/Config"
+  /config/datastore:
```
Comment (mathnogueira) [inline api/openapi.yaml:1076]: Should the `get` method be included in this PR? It's ok if the answer is 'no'.

**Review by mathnogueira — CHANGES_REQUESTED** [review]
Small thing about the openapi spec

File: api/config.yaml (hunk 1 above)
Comment (xoscar) [reply api/config.yaml:64]: Great callout  @mathnogueira in that case, should we include a name entry or something like that?

**Review by mathnogueira — COMMENTED** [review]
OpenAPI spec looks good

**Comment by xoscar** [comment]
@mathnogueira @jorgeepc this is ready to be re-reviewed, if you think it looks good please approve so we can merge and move forward. Thanks guys!

**Comment by xoscar** [comment]
@mathnogueira this PR is kind of a blocker for the FE, if you think everything looks fine, do you mind approving?

**Review by mathnogueira — APPROVED** [review]

**Pull request merged**